src/Deprecated/Engine/Engine/IntrinsicFunctions.cs

```diff
@@ -197,26 +197,34 @@ internal static object GetRegistryValueFromView(string keyName, string valueName
                     // of that error.
                     RegistryView view = (RegistryView)Enum.Parse(typeof(RegistryView), viewAsString, true);
 
-                    using (RegistryKey key = GetBaseKeyFromKeyName(keyName, view, out subKeyName))
+                    RegistryKey key = null;
+                    try
                     {
-                        if (key != null)
+                        using (key = GetBaseKeyFromKeyName(keyName, view, out subKeyName))
                         {
-                            using (RegistryKey subKey = key.OpenSubKey(subKeyName, false))
+                            if (key != null)
                             {
-                                // If we managed to retrieve the subkey, then move onto locating the value
-                                if (subKey != null)
+                                using (RegistryKey subKey = key.OpenSubKey(subKeyName, false))
                                 {
-                                    result = subKey.GetValue(valueName);
-                                }
-
-                                // We've found a value, so stop looking
-                                if (result != null)
-                                {
-                                    break;
+                                    // If we managed to retrieve the subkey, then move onto locating the value
+                                    if (subKey != null)
+                                    {
+                                        result = subKey.GetValue(valueName);
+                                    }
+
+                                    // We've found a value, so stop looking
+                                    if (result != null)
+                                    {
+                                        break;
+                                    }
                                 }
                             }
                         }
                     }
+                    finally
+                    {
+                        key?.Dispose();
```

Comment:
This one I don't understand, what was wrong with it before?

Reply:
![image](https://github.com/dotnet/msbuild/assets/95473390/7676e38a-8a47-4abf-b822-8aff3a753f1e)

Reply:
Same as the other occurrence of this pattern, I think it's worth reporting as an analyzer bug.